src/sql/src/plan/statement/ddl.rs

```diff
@@ -3403,6 +3403,8 @@ pub fn plan_create_connection(
     let connection = connection_options_extracted.try_into_connection(scx, connection_type)?;
     if let Connection::Aws(_) = &connection {
         scx.require_feature_flag(&vars::ENABLE_AWS_CONNECTION)?;
+    } else if let Connection::MySql(_) = &connection {
+        scx.require_feature_flag(&vars::ENABLE_MYSQL_CONNECTION)?;
```

Comment:
let's use a general named flag for the whole MySQL feature that everything will be gated with